### PR TITLE
Sound Null Safety

### PR DESCRIPTION
**Pull request by diegogarciar**
Hello, I wanted to contribute by adding null safety to this package, I used mainly the migration tool and also removed added some `required` for fields that were always initialized as non-null. I also optimized some checks to use optional chaining instead of null checking.

### CODE REVIEW

**Comment by fayeed** [comment]
@diegogarciar Also if you can also update flutter_parsed_text to v2.1.0 that also support null safety that would be great.

File: pubspec.yaml [hunk 1]
```diff
@@ -4,15 +4,15 @@ version: 1.1.15
...
+  sdk: '>=2.12.0 <3.0.0'
 
 dependencies:
   flutter:
     sdk: flutter
-  flutter_parsed_text: ^1.1.0
-  uuid: ^2.0.1
-  intl: ^0.16.0
-  transparent_image: ^1.0.0
+  flutter_parsed_text: ^2.1.0
```
Comment (diegogarciar) [inline pubspec.yaml:12]: @fayeed its already at 2.1.0

Comment (fayeed) [reply pubspec.yaml:12]: sure my bad, I must have seen something else

File: lib/src/widgets/load_earlier.dart [hunk 2]
```diff
@@ -2,24 +2,20 @@ part of dash_chat;
...
-    Key key,
-    @required this.onLoadEarlier,
-    @required this.defaultLoadCallback,
+    Key? key,
+    required this.onLoadEarlier,
+    required this.defaultLoadCallback,
   }) : super(key: key);
 
-  final Function onLoadEarlier;
+  final Function? onLoadEarlier;
```
Comment (fayeed) [inline lib/src/widgets/load_earlier.dart:10]: If it's required then the variable can't be non-nullable

Comment (diegogarciar) [reply lib/src/widgets/load_earlier.dart:10]: nice catch!

**Comment by fayeed** [comment]
@diegogarciar Thanks for the PR

**Comment by fayeed** [comment]
@diegogarciar Updated the package to `1.1.15-nullsafety`

**Pull request merged**